engine/table/src/main/java/io/deephaven/engine/util/PythonScopeJpyImpl.java

```diff
@@ -189,7 +189,16 @@ public static Object convert(PyObject pyObject) {
         } else if (pyObject.isCallable()) {
             return wrapCallable(pyObject);
         } else if (pyObject.isConvertible()) {
-            return pyObject.getObjectValue();
+            Object objValue = pyObject.getObjectValue();
+            // workaround a JPY issue with determining the correct Java type a Python 'int' should be
+            // mapped to (int or long) without causing an implicit overflow
+            if (objValue instanceof  Integer || objValue.getClass() == int.class) {
+                long longValue;
+                if ((longValue = pyObject.getLongValue()) > (int) objValue) {
```

Comment:
I think `!=` is more appropriate?

Note: we'll still have issues w/ python "ints" that really should be BigInteger (> 64 bits)

Reply:
!= is definitely the correct way, covering both negative and positive values; and yes, we need to address the BigDecimal translation in the future.